### PR TITLE
Fix provisioning script and enable CI tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/COM-560

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ before_script:
   - docker build -t $APP_IMAGE .
   - docker pull quay.io/keboola/azure-cli
   # create synapse server
-  - . ./provisioning/synapse/synapse.sh -c
+  - SYNAPSE_ENV=`./provisioning/synapse/synapse.sh -c` && export $(echo ${SYNAPSE_ENV} | xargs)
```
Comment (michaljurecko) [inline .travis.yml:14]: Script sa tu sourcoval, no nebolo mozne spravit, aby v pripade chyby spadol (kedze modifikoval Travis env).
Preto sa teraz spusta standardne, ako samostatny proces, (na STDOUT sa zapisu exporty) a potom sa to exportuje v dalsom kroku.

File: composer.json [hunk 2]
```diff
@@ -44,7 +44,8 @@
         "build": [
             "@phplint",
             "@phpcs",
-            "@phpstan"
+            "@phpstan",
+            "@tests"
```
Comment (michaljurecko) [inline composer.json:48]: Mali sme tu vypnute testy, lebo sme nemali premigrovanu Synapse pod nasu subskripciu. Fixnute.

(Vyplo sa to tu: https://github.com/keboola/synapse-transformation/pull/7)

File: provisioning/synapse/synapse.sh [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+#set -o xtrace          # Trace the execution of the script (debug)
 
-#REQUIRED ENV VARS
-#SYNAPSE_SERVER_NAME=
-#AZURE_RESOURCE_GROUP=
-#AZURE_SERVICE_PRINCIPAL_TENANT=
-#AZURE_SERVICE_PRINCIPAL=
-#AZURE_SERVICE_PRINCIPAL_PASSWORD=
+# Required environment variables
+: "${SYNAPSE_SERVER_NAME:?Need to set SYNAPSE_SERVER_NAME env variable}"
```
Comment (michaljurecko) [inline provisioning/synapse/synapse.sh:10]: Premenne sa kontroluju na zaciatku scriptu.

File: provisioning/synapse/synapse.sh [hunk 4]
```diff
@@ -1,15 +1,19 @@
 #!/usr/bin/env bash
 
+set -o errexit          # Exit on most errors (see the manual)
```
Comment (michaljurecko) [inline provisioning/synapse/synapse.sh:3]: Ak spadne nejaka cast scriptu, tak to spadne cele.

File: provisioning/synapse/synapse.sh [hunk 5]
```diff
@@ -44,41 +47,42 @@ createServer(){
     export SYNAPSE_DW_SERVER_NAME=$(echo ${output} | jq -r '.properties.outputs.warehouseName.value')
     export SYNAPSE_RESOURCE_ID=$(echo ${output} | jq -r '.properties.outputs.warehouseResourceId.value')
 
-    echo "Server deployed."
-    echo $SYNAPSE_SQL_SERVER_NAME
-    echo $SYNAPSE_DW_SERVER_NAME
-    echo $SYNAPSE_RESOURCE_ID
+    # Log messages to STDERR, so STDOUT contains only exports
+    echo "Server deployed: $SYNAPSE_SQL_SERVER_NAME" 1>&2;
```
Comment (michaljurecko) [inline provisioning/synapse/synapse.sh:55]: Hlasky zapisujeme na STDERR, aby na STDOUT boli iba exporty.

File: provisioning/synapse/synapse.sh [hunk 6]
```diff
@@ -44,41 +47,42 @@ createServer(){
...
+    echo "Firewall rule set." 1>&2;
+    echo "$output" 1>&2;
 
-#    Set vars for php app
-    export SYNAPSE_UID=keboola
-    export SYNAPSE_PWD=${SYNAPSE_SERVER_PASSWORD}
-    export SYNAPSE_DATABASE=${SYNAPSE_DW_SERVER_NAME}
-    export SYNAPSE_SERVER=${SYNAPSE_SQL_SERVER_NAME}.database.windows.net
+    # Set vars for php app
+    echo "export SYNAPSE_UID=keboola"
```
Comment (michaljurecko) [inline provisioning/synapse/synapse.sh:72]: Exporty.

File: provisioning/synapse/synapse.sh [hunk 7]
```diff
@@ -44,41 +47,42 @@ createServer(){
...
+    echo "export SYNAPSE_PWD=${SYNAPSE_SERVER_PASSWORD}"
+    echo "export SYNAPSE_DATABASE=${SYNAPSE_DW_SERVER_NAME}"
+    echo "export SYNAPSE_SERVER=${SYNAPSE_SQL_SERVER_NAME}.database.windows.net"
 }
 
 deleteServer(){
+  : "${SYNAPSE_DW_SERVER_NAME:?Need to set SYNAPSE_DW_SERVER_NAME env variable (exported by the create cmd).}"
```
Comment (michaljurecko) [inline provisioning/synapse/synapse.sh:82]: Tieto premenne vytvara `create` cmd, a musia byt pritomne pri delete.

File: tests/functional/query-select/expected/data/out/db-dump/new-table.metadata2.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
+[
+      {
```
Comment (michaljurecko) [inline tests/functional/query-select/expected/data/out/db-dump/new-table.metadata2.json]: V https://github.com/keboola/synapse-transformation/pull/7 sme pridali podporu pre `SELECT INTO` -> aby sa spustal, ... no test sme nefixli spravne, chybala tu tato tabulka, ktora sa vytvrara tu: https://github.com/keboola/synapse-transformation/blob/c4c9eb4b62926e05450b74486bb809615c41a710/tests/functional/query-select/source/data/config.json#L21

**Review by michaljurecko — COMMENTED** [review]
@ondrajodas mohol by si sa prosim na to pozriet, CSAS dostava query timeout, a aby som to mohol upravit, tak som v tomto PR musel rozbehat CIcko.

- CI Synapse je uz premigrovana do nasej Subskripcie.
- FIxol som synapse provisioning script, aby ak je nejaka chyba spadol cely build.
- Fixol som `select-query` datadir test, ktory zostal nefunkcny po tom ako sme minule vyply testy.

**Comment by michaljurecko** [comment]
Linky na build:
- Vytvorenie Synapse: https://travis-ci.com/github/keboola/synapse-transformation/builds/206518561#L1036
- Unit testy: https://travis-ci.com/github/keboola/synapse-transformation/builds/206518561#L1075
- Datadir testy: https://travis-ci.com/github/keboola/synapse-transformation/builds/206518561#L1086
- Zmazanie Synapse: https://travis-ci.com/github/keboola/synapse-transformation/builds/206518561#L1348

**Review by ondrajodas — APPROVED** [review]
🎉

**Pull request merged**